zephyr/Kconfig

```diff
@@ -96,6 +96,16 @@ if MENDER_MCU_CLIENT
 
     endmenu
 
+    menu "Update modules configuration"
+
+        config MENDER_CLIENT_ZEPHYR_IMAGE_UPDATE_MODULE
+            bool "The default Mender zephyr-image update module"
+            default y
+            help
+                The default zephyr-image update module handles full Zephyr images on MCUboot-based devices.
+
```

Comment:
There are a set of dependencies that are only needed if this configuration is on, so we should move here at least:
```
    select BOOTLOADER_MCUBOOT
    select IMG_ENABLE_IMAGE_CHECK
    select IMG_ERASE_PROGRESSIVELY
    select IMG_MANAGER
    select MPU_ALLOW_FLASH_WRITE
    select STREAM_FLASH
```

Reply:
Are you sure the `MPU_ALLOW_FLASH_WRITE` and `STREAM_FLASH` are not required for NVS?

Reply:
> Are you sure the `MPU_ALLOW_FLASH_WRITE` 

This one I copied my mistake.

> and `STREAM_FLASH` are not required for NVS?

It is [not required](https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_STREAM_FLASH), as far as I can tell.